compaction.go

```diff
@@ -1,16 +1,55 @@
 package lsm
 
 func (db *DB) compactLevel(level int) error {
-	if level == 0 {
+	if level == 0 && db.needL0Compaction() {
```

Comment:
### Prerequisite Checks
Ensure that `db.needL0Compaction()` and `db.needLevelNCompaction(level)` are not nil before calling them. This can be done by adding a nil check:
```go
if db != nil && level == 0 && db.needL0Compaction() {
```
Similarly, for other methods like `db.compactLevelN(level)` and `db.involvedIterators(level+1, minKey, maxKey)`.